src/components/app/pagePolls/pollContainer.tsx

```diff
@@ -0,0 +1,55 @@
+'use client'
+
+import { createContext, ReactNode, useCallback } from 'react'
+
+import { PollResultsDataResponse, PollsVotesFromUserResponse } from '@/data/polls/getPollsData'
+import { usePollsResultsData } from '@/hooks/usePollsResultsData'
+import { usePollsVotesFromUser } from '@/hooks/usePollsVotesFromUser'
+import { useSession } from '@/hooks/useSession'
+
+interface PollContainerProps {
+  children: ReactNode
+  pollsResultsData: Record<string, PollResultsDataResponse>
+}
+
+export const PollContainerContext = createContext<{
```

Comment:
Loved it, would only change a couple of details:
- The name doesn't make sense, this is a context for the Poll page, not for the container
- I'd suggest creating a separate file to encapsulate this `createContext` call and I'd also create a helper hook to make the call simpler and also make sure we're not calling it outside of the page context, something like this

```suggestion
// pollPageContext.ts

interface PollPageContextValue {
  isLoading: boolean
  userPolls: PollsVotesFromUserResponse | undefined
  pollsResults: Record<string, PollResultsDataResponse>
  handleRefreshVotes: () => Promise<void>
}

const PollPageContext = createContext<PollPageContextValue | null>(null)

export function usePollPageContext() {
  const contextValue = useContext(PollPageContext)

  if (!contextValue) {
    throw new Error('usePollPageContext called outside of PollPageContext') 
  }
  
  return contextValue
}
```